miner/hash_blocklist.go

```diff
@@ -0,0 +1,20 @@
+package miner
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type HashBlocklist struct {
+	*Blocklist[common.Hash, types.Header]
+}
+
+var TxHashEvictionTimeoutSeconds uint64 = 24 * hours
+
+func TxHashReachedEvictionTimeout(h *types.Header, latest *types.Header) bool {
+	return h.Time+TxHashEvictionTimeoutSeconds < latest.Time
```

Comment:
if i understand correctly you're storing the whole header for each TX, which makes the blocklist rather big in memeroy, i'd store just the eviction date.. or something..

keep it as simple/small as possible